unifai/agent/utils.py

```diff
@@ -53,26 +54,38 @@ def generate_uuid_from_id(id_str: str) -> uuid.UUID:
     """Generate a UUID from a string identifier."""
     return uuid.uuid5(uuid.NAMESPACE_DNS, id_str)
 
-def get_collection_name(agent_id: str, client_id: str, chat_id: str) -> str:
-    """Generate a unique collection name for memory storage"""
-    collection_base = f"{agent_id}-{client_id}-{chat_id}"
-    sanitized = ''.join(c if c.isalnum() else '-' for c in collection_base)
-    if not sanitized[0].isalpha():
-        sanitized = 'id-' + sanitized
-    if len(sanitized) < 3:
-        sanitized = sanitized + '-collection'
-    elif len(sanitized) > 63:
-        sanitized = sanitized[:60] + '-col'
-    return sanitized
+def get_collection_name(agent_id: str, user_id: str, chat_id: str = None) -> str:
+    
+    if not chat_id:
+        chat_id = user_id
+    
+    if agent_id:
+        base_name = f"{agent_id}-{user_id[:8]}-{chat_id[:8]}"
```

Comment:
we shouldn't truncate the user id or chat id here as it might cause name collision